make/dev.mk

```diff
@@ -1,7 +1,11 @@
 ##@ Development
 
+.PHONY: agents-templates
+agents-templates: ## Copy agents' templates for Primaza's Control Plane deployment mechanism
+	cp config/agents/app/agent/agentapp.yaml config/agents/svc/agent/agentsvc.yaml pkg/primaza/workercluster/templates/
```

Comment:
Don't we need to build them with `kustomize` first?  This is how we would inject dependencies, e.g. setting up cert-manager in the deployment.

Reply:
Maybe something like this?

<details><summary>Details</summary>
<p>

```patch
From b8bee0fffe776216437f1dff3bce12f9b07acf82 Mon Sep 17 00:00:00 2001
From: Andy Sadler <ansadler@redhat.com>
Date: Fri, 5 May 2023 14:21:21 -0500
Subject: [PATCH] generate and embed deployments from kustomize

The deployments we embed and push to worker clusters need to be
processed with kustomize.

Signed-off-by: Andy Sadler <ansadler@redhat.com>
---
 make/deps.mk                                  |  7 ++
 make/dev.mk                                   |  5 +-
 .../workercluster/templates/.gitignore        |  2 +
 .../workercluster/templates/agentapp.yaml     | 62 ----------------
 .../workercluster/templates/agentsvc.yaml     | 71 -------------------
 5 files changed, 12 insertions(+), 135 deletions(-)
 create mode 100644 pkg/primaza/workercluster/templates/.gitignore
 delete mode 100644 pkg/primaza/workercluster/templates/agentapp.yaml
 delete mode 100644 pkg/primaza/workercluster/templates/agentsvc.yaml

diff --git a/make/deps.mk b/make/deps.mk
index 08ff4f2..d1a063a 100644
--- a/make/deps.mk
+++ b/make/deps.mk
@@ -9,11 +9,13 @@ $(LOCALBIN):
 KUSTOMIZE ?= $(LOCALBIN)/kustomize
 CONTROLLER_GEN ?= $(LOCALBIN)/controller-gen
 ENVTEST ?= $(LOCALBIN)/setup-envtest
+YQ ?= $(LOCALBIN)/yq
 
 ## Tool Versions
 KUSTOMIZE_VERSION ?= v4.5.7
 CONTROLLER_TOOLS_VERSION ?= v0.11.3
 CERTMANAGER_VERSION ?= v1.11.1
+YQ_VERSION ?= v4
 
 KUSTOMIZE_INSTALL_SCRIPT ?= "https://raw.githubusercontent.com/kubernetes-sigs/kustomize/master/hack/install_kustomize.sh"
 .PHONY: kustomize
@@ -48,6 +50,11 @@ OPM = $(shell which opm)
 endif
 endif
 
+.PHONY: yq
+yq: $(YQ) ## Download envtest-setup locally if necessary.
+$(YQ): $(LOCALBIN)
+	test -s $(YQ) || GOBIN=$(LOCALBIN) $(GO) install github.com/mikefarah/yq/v4@$(YQ_VERSION)
+
 .PHONY: deploy-cert-manager
 deploy-cert-manager:
 	kubectl apply -f https://github.com/cert-manager/cert-manager/releases/download/$(CERTMANAGER_VERSION)/cert-manager.yaml
diff --git a/make/dev.mk b/make/dev.mk
index a6d06b6..2605c44 100644
--- a/make/dev.mk
+++ b/make/dev.mk
@@ -1,8 +1,9 @@
 ##@ Development
 
 .PHONY: agents-templates
-agents-templates: ## Copy agents' templates for Primaza's Control Plane deployment mechanism
-	cp config/agents/app/agent/agentapp.yaml config/agents/svc/agent/agentsvc.yaml pkg/primaza/workercluster/templates/
+agents-templates: kustomize yq ## Copy agents' templates for Primaza's Control Plane deployment mechanism
+	@$(KUSTOMIZE) build --load-restrictor LoadRestrictionsNone config/agents/app/default/ | $(YQ) eval 'select(.kind == "Deployment" and .apiVersion == "apps/v1")' > pkg/primaza/workercluster/templates/agentapp.yaml
+	@$(KUSTOMIZE) build --load-restrictor LoadRestrictionsNone config/agents/svc/default/ | $(YQ) eval 'select(.kind == "Deployment" and .apiVersion == "apps/v1")' > pkg/primaza/workercluster/templates/agentsvc.yaml
 
 .PHONY: manifests
 manifests: agents-templates controller-gen ## Generate WebhookConfiguration, ClusterRole and CustomResourceDefinition objects.
diff --git a/pkg/primaza/workercluster/templates/.gitignore b/pkg/primaza/workercluster/templates/.gitignore
new file mode 100644
index 0000000..509e354
--- /dev/null
+++ b/pkg/primaza/workercluster/templates/.gitignore
@@ -0,0 +1,2 @@
+agentapp.yaml
+agentsvc.yaml
diff --git a/pkg/primaza/workercluster/templates/agentapp.yaml b/pkg/primaza/workercluster/templates/agentapp.yaml
deleted file mode 100644
index ddbe28e..0000000
--- a/pkg/primaza/workercluster/templates/agentapp.yaml
+++ /dev/null
@@ -1,62 +0,0 @@
-apiVersion: apps/v1
-kind: Deployment
-metadata:
-  name: primaza-app-agent
-  labels:
-    app.kubernetes.io/part-of: primaza
-    control-plane: primaza-app-agent
-  finalizers:
-  - agentapp.primaza.io/finalizer
-spec:
-  selector:
-    matchLabels:
-      control-plane: primaza-app-agent
-  replicas: 1
-  template:
-    metadata:
-      annotations:
-        kubectl.kubernetes.io/default-container: manager
-      labels:
-        control-plane: primaza-app-agent
-    spec:
-      securityContext:
-        runAsNonRoot: true
-      containers:
-      - command:
-        - /manager
-        args:
-        - --leader-elect
-        image: agentapp:latest
-        imagePullPolicy: IfNotPresent
-        name: manager
-        env:
-          - name: WATCH_NAMESPACE
-            valueFrom:
-              fieldRef:
-                fieldPath: metadata.namespace
-        securityContext:
-          allowPrivilegeEscalation: false
-          capabilities:
-            drop:
-              - "ALL"
-        livenessProbe:
-          httpGet:
-            path: /healthz
-            port: 8081
-          initialDelaySeconds: 15
-          periodSeconds: 20
-        readinessProbe:
-          httpGet:
-            path: /readyz
-            port: 8081
-          initialDelaySeconds: 5
-          periodSeconds: 10
-        resources:
-          limits:
-            cpu: 500m
-            memory: 128Mi
-          requests:
-            cpu: 10m
-            memory: 64Mi
-      serviceAccountName: primaza-app-agent
-      terminationGracePeriodSeconds: 10
diff --git a/pkg/primaza/workercluster/templates/agentsvc.yaml b/pkg/primaza/workercluster/templates/agentsvc.yaml
deleted file mode 100644
index 9797be4..0000000
--- a/pkg/primaza/workercluster/templates/agentsvc.yaml
+++ /dev/null
@@ -1,71 +0,0 @@
-apiVersion: apps/v1
-kind: Deployment
-metadata:
-  name: primaza-svc-agent
-  labels:
-    app.kubernetes.io/part-of: primaza
-    control-plane: primaza-svc-agent
-  finalizers:
-    - agent.primaza.io/finalizer
-spec:
-  selector:
-    matchLabels:
-      control-plane: primaza-svc-agent
-  replicas: 1
-  template:
-    metadata:
-      annotations:
-        kubectl.kubernetes.io/default-container: manager
-      labels:
-        control-plane: primaza-svc-agent
-    spec:
-      securityContext:
-        runAsNonRoot: true
-      containers:
-      - command:
-        - /manager
-        args:
-        - --leader-elect
-        image: agentsvc:latest
-        imagePullPolicy: IfNotPresent
-        name: manager
-        env:
-          - name: WATCH_NAMESPACE
-            valueFrom:
-              fieldRef:
-                fieldPath: metadata.namespace
-        securityContext:
-          allowPrivilegeEscalation: false
-          capabilities:
-            drop:
-              - "ALL"
-        livenessProbe:
-          httpGet:
-            path: /healthz
-            port: 8081
-          initialDelaySeconds: 15
-          periodSeconds: 20
-        readinessProbe:
-          httpGet:
-            path: /readyz
-            port: 8081
-          initialDelaySeconds: 5
-          periodSeconds: 10
-        resources:
-          limits:
-            cpu: 500m
-            memory: 128Mi
-          requests:
-            cpu: 10m
-            memory: 64Mi
-        volumeMounts:
-        - mountPath: /tmp/k8s-webhook-server/serving-certs
-          name: cert
-          readOnly: true
-      serviceAccountName: primaza-svc-agent
-      terminationGracePeriodSeconds: 10
-      volumes:
-      - name: cert
-        secret:
-          defaultMode: 420
-          secretName: webhook-server-cert
-- 
2.40.1

```

</p>
</details>

Reply:
great idea, thanks!